pkg/util/diff.go

```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2025 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package util
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/go-test/deep"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// DiffResult is the interface that represents the result of a diff operation.
+type DiffResult interface {
+	Changed() bool
+	String() string
+}
+
+type diffResult struct {
+	diff []string
+}
+
+// Changed returns true if the diff detected any changes.
+func (d *diffResult) Changed() bool {
+	return len(d.diff) > 0
+}
+
+// String returns the diff as a string.
+func (d *diffResult) String() string {
+	if !d.Changed() {
+		return ""
+	}
+
+	out := "." + strings.Join(d.diff, ", .")
+	out = strings.ReplaceAll(out, ".slice[", "[")
+	out = strings.ReplaceAll(out, "map[", "[")
```

Comment:
Might be a nit, but could we distinguish maps from slices with `{}`? Might be too much work, since we'd have to find and replace the closing characters, but it could be helpful in interpreting output.

Reply:
Main reason why I did the [] thing was e.g. label keys, because they have dots.